### DISCUSSION TITLE
Use dark mode as a basis for custom themes

### DISCUSSION THREAD

**Discussion by noxhirsch** [home-assistant/frontend#8031] [Theming] [answered]
At the moment, if you want to have a custom dark theme, you have to change nearly every theme-variable to a dark color. 

It would be way easier (and with less breaking changes if some variables change in the future), if a theme could use the default dark theme as a basis for a custom dark theme.
Especially if you only want to change e.g. the header or background color of the default dark theme you then would only have to set 1-2 variables instead of ~70 (rough estimate)

As an example, this could be used in the theme file to define if it's a dark theme or not:
```
my_fancy_dark_theme:
  mode: dark
  primary-background-color: '#550000'
```

**Comment by benganellison** [comment]
@bramkragten how much effort would you say that is it to make this happen? I would like so see if i can contribute with this and where in the code should i start looking? i would prefer to work on this rather then on a personal dark theme

**Reply by spacegaier** [reply]
> 
> 
> @bramkragten how much effort would you say that is it to make this happen? I would like so see if i can contribute with this and where in the code should i start looking? i would prefer to work on this rather then on a personal dark theme

It is not too trivial since it also impacts the backend. I discussed this topic today a bit with Bram and I am working on a PR.

**Comment by KTibow** [comment]
This might help in the meantime:
https://github.com/SilvrrGIT/Home-Assistant-Themes/blob/master/themes/default_ha.yaml
https://community.home-assistant.io/t/exercise-recreate-the-default-dark-theme/235948

**Comment by spacegaier** [comment, marked as answer]
FYI: I have a working solution, that I will propose as PR: https://github.com/home-assistant/frontend/pull/8347

**Reply by noxhirsch** [reply]
Thank you very much for your effort - it's perfect! 😊
